src/commands/containers/composeGroup.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IActionContext } from 'vscode-azureextensionui';
+import { localize } from '../../localize';
+import { ContainerGroupTreeItem } from '../../tree/containers/ContainerGroupTreeItem';
+import { ContainerTreeItem } from '../../tree/containers/ContainerTreeItem';
+import { executeAsTask } from '../../utils/executeAsTask';
+import { rewriteCommandForNewCliIfNeeded } from '../compose';
+
+export async function composeGroupRestart(context: IActionContext, node: ContainerGroupTreeItem): Promise<void> {
+    return composeGroup(context, 'restart', node);
+}
+
+export async function composeGroupDown(context: IActionContext, node: ContainerGroupTreeItem): Promise<void> {
+    return composeGroup(context, 'down', node);
+}
+
+async function composeGroup(context: IActionContext, composeCommand: 'restart' | 'down', node: ContainerGroupTreeItem): Promise<void> {
+    const workingDirectory = getComposeWorkingDirectory(node);
+    const filesArgument = getComposeFiles(node)?.map(f => `-f ${f}`)?.join(' ');

```

Comment:
These file names should probably be quoted, to account for cases like spaces in the name, or the file name beginning with a dash 🙄

Reply:
Hm, is `'` reliable in all shells or should we use `"`? Or is that also bust?

Reply:
I believe the rule of thumb is single quote on Mac/Linux; double quote on Windows https://github.com/microsoft/vscode-docker/blob/master/src/commands/dockerInstaller.ts#L85